### ISSUE TITLE
Add quick action to check whether the CS building is open

### ISSUE THREAD

**Issue by CarrotManMatt** [CSSUoB/TeX-Bot-Py-V2#112]
Using either webscraping or the API (if it gets developed) of https://iscsopen.com TeXBot should provide information on whether the CS building is open.

A credits message should also be added to mention that the information is retrieved from https://iscsopen.com.

Suggestions/reasons on whether or not to include this feature are welcomed.

**Comment by codemicro** [comment]
Matt you do ***not*** need an API for this.

```js
var currentTime = new Date();
            var currentHour = currentTime.getHours();
            var currentDay = currentTime.getDay();

            var isOpen = false;

            if (studentType === 'non-taught' || (studentType === 'taught' && (currentHour >= 9 && currentHour < 23 && currentDay >= 1 && currentDay <= 5) || (currentHour >= 9 && currentHour < 17 && (currentDay === 0 || currentDay === 6)))) {
                isOpen = true;
            }
```

**Comment by sapphyree** [comment]
> Matt you do _**not**_ need an API for this.
> 
> ```js
> var currentTime = new Date();
> var currentHour = currentTime.getHours();
> var currentDay = currentTime.getDay();
> 
> var isOpen = false;
> 
> if (studentType === 'non-taught' || (studentType === 'taught' && (currentHour >= 9 && currentHour < 23 && currentDay >= 1 && currentDay <= 5) || (currentHour >= 9 && currentHour < 17 && (currentDay === 0 || currentDay === 6)))) {
>     isOpen = true
> }
> ```

Whilst I agree here, I'd recommend the removal of magic numbers + the consolidation of these conditionals to constants that are defined outside of the main conditional for clarity.

i.e: `IS_NON_TAUGHT_STUDENT = studentType === 'non-taught'`

**Comment by CarrotManMatt** [comment]
> Matt you do _**not**_ need an API for this.
> 
> ```js
> var currentTime = new Date();
> var currentHour = currentTime.getHours();
> var currentDay = currentTime.getDay();
> 
> var isOpen = false;
> 
> if (studentType === 'non-taught' || (studentType === 'taught' && (currentHour >= 9 && currentHour < 23 && currentDay >= 1 && currentDay <= 5) || (currentHour >= 9 && currentHour < 17 && (currentDay === 0 || currentDay === 6)))) {
>     isOpen = true
> }
> ```

Admittedly the solution is not very complex, however I do not want to have to deal with updating the rules for edge-cases & holidays, especially when an existing solution exists (https://iscsopen.com), which is actively maintained by someone else

**Comment by codemicro** [comment]
No offence to Dak, but I highly doubt that iscsopen is going to be accurate long-term.

**Comment by CarrotManMatt** [comment]
However, I expect `get_effort(DAK) > get_effort(MATT)`. So for the mid-term it would be?

**Comment by FrozenDude101** [comment]
Any version of this isn't going to be accurate long term.
There isn't an API/website for this that isn't manually updated to include open/close dates.

**Comment by CarrotManMatt** [comment]
> Any version of this isn't going to be accurate long term. There isn't an API/website for this that isn't manually updated to include open/close dates.

Agreed. Therefore I would be more willing to rely on a developer, whose sole focus is ensuring the opening-times accuracy, rather than a developer that has many other [issues](https://github.com/CSSUoB/TeX-Bot-Py-V2/issues) to deal with (🥲).

**Comment by CarrotManMatt** [comment]
I was once again proved wrong, and https://iscsopen.com has now gone offline. This feature should have the calculation performed internally by TexBot.
